author	Claudio Takahasi <claudio.takahasi@openbossa.org>
commit	bd379384451d725c0df547708c3a746599c9375a
	Tue, 3 Apr 2012 11:18:47 +0000 (14:18 +0300)

attrib: Fix missing Discovery reply when disconnected

This patch fixes missing reply for DiscoverCharacteristics method on
Generic Attribute. If link is disconnected reply message is not sent.

diff -cr bluez-4.99.pan.orig/attrib/client.c bluez-4.99.pan/attrib/client.c
*** bluez-4.99.pan.orig/attrib/client.c	Mon Oct 15 02:52:10 2012
--- bluez-4.99.pan/attrib/client.c	Mon Oct 15 03:01:32 2012
***************
*** 332,337 ****
--- 332,351 ----
  {
  	struct gatt_service *gatt = user_data;
  
+ 	if (gatt->query && gatt->query->msg) {
+ 		DBusMessage *reply;
+ 
+ 		reply = btd_error_failed(gatt->query->msg,
+ 					"ATT IO channel was disconnected");
+ 		g_dbus_send_message(gatt->conn, reply);
+ 		dbus_message_unref(gatt->query->msg);
+ 	}
+ 
+ 	if (gatt->query) {
+ 		g_slist_free_full(gatt->query->list, g_free);
+ 		gatt->query = NULL;
+ 	}
+ 
  	if (gatt->attrib) {
  		g_attrib_unref(gatt->attrib);
  		gatt->attrib = NULL;
